### PR TITLE
Ryan dev4 - Big Update 

### PR DESCRIPTION
**Pull request by ryanwooa**
1. Update SingleItem
2. Update Todolist
3. Create Icon component: use this compoent to show the icons.
4. Create PopupMenu component: any pop-up menu can use this compoent ,include show/hide function and Auto-calculate display position
5. Create CheckBoxButton component for colored checkbox with different priority
6. Update Filter-lables page: part of ffunction need to be finished
7. Update Project page: different boards' tasks could display in different section
!!!8. Adjust Modal( already merged from develop branch, and there is small problem with Lyvia's version), Please prioritize my version. Will discuss in this week meeting.
9. Update Topbar (update application layout)
10. Part of CSS adjustment
11. Update function of add project
12. Update Leftbar

### CODE REVIEW

File: app/application/[menu]/[submenu]/page.js [hunk 1]
```diff
@@ -16,14 +18,28 @@ export default async function SubAppPages({ params }) {
     }
   }
   const tasks = await getOneUserTasks();
```
Comment (zli213) [inline app/application/[menu]/[submenu]/page.js:20]: I haven't see you import this method file, so how could it work? And in`src/utils/data`, I saw a file called 'getOneUserTasks.js', is it ? Btw, there is params in this method, is it ok when using getOneUserTasks(empty here)?

**Review by zli213 — COMMENTED** [review]
Sorry, I misunderstood earlier. I'll take a closer look later and then give my comments.

File: components/application/widgets/TodoList.js [hunk 2]
```diff
@@ -1,31 +1,88 @@
 // TodoList.js
 // "use client";
```
Comment (zli213) [inline components/application/widgets/TodoList.js:2]: Delete this one.

**Review by zli213 — COMMENTED** [review]
I test this branch in local environment and I didn't find bugs.

File: components/application/widgets/TodoList.js (hunk 2 above)
Comment (ryanwooa) [reply components/application/widgets/TodoList.js]: Deleted

**Review by kesesek — COMMENTED** [review]
No questions

File: components/application/widgets/Scheduler.js [hunk 3]
```diff
@@ -451,6 +461,46 @@ export function formatDate(inDate) {
   return dateJson;
 }
 
+function convertPosition(position) {
```
Comment (jlmjkfd) [inline components/application/widgets/Scheduler.js:464]: Some other components may use this function, such as priority selector, and tag selector. This function can be put in a public library.

**Review by jlmjkfd — APPROVED** [review]
No obvious problems.

**Review by zli213 — APPROVED** [review]
It's ok

File: components/application/widgets/Scheduler.js (hunk 3 above)
Comment (ryanwooa) [reply components/application/widgets/Scheduler.js]: Ok, It has been placed in a new file

**Pull request merged**